htdocs/public/payment/paymentok.php

```diff
@@ -412,7 +412,13 @@
 
 				$datesubend = null;
 				if ($datesubscription && $defaultdelay && $defaultdelayunit) {
-					$datesubend = dol_time_plus_duree(dol_time_plus_duree($datesubscription, $defaultdelay, $defaultdelayunit), -1, 'd');
+					$datesubend = dol_time_plus_duree($datesubscription, $defaultdelay, $defaultdelayunit);
+					// the new end date of subscription must be in futur
```

Comment:
Not necessarly. When a subscription is done and is was expired since 10 days for example, the subscription for the year must restart from the end date of previous subscription, so 10 days ago.

Reply:
Yes but the end date should be in future.

That is the purpose of the while as it loop on the end sub date.

Reply:
I see.
Why 
while (dol_time_plus_duree($datesubend, -1, 'd') < $now) {
and not
while ($datesubend < $now) {
removing 1 day into a look that add a delay that may be 1 day to (even if commonly it is 1 year) may generate an infinite loop

Reply:
Much better. Thx

Reply:
> I see.
> Why 
> while (dol_time_plus_duree($datesubend, -1, 'd') < $now) {
> and not
> while ($datesubend < $now) {
> removing 1 day into a look that add a delay that may be 1 day to (even if commonly it is 1 year) may generate an infinite loop

I'll change that, I used this cause it was the old way to define the end date